docs/linking/README.md

```diff
@@ -1,3 +1,255 @@
+
+![Authentication (OTP)](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/mojaloop/pisp/master/docs/linking/3-authentication-otp.puml)
+
+## 1.5. Grant consent
+
+Now that mutual trust has been established between all three parties, the DFSP
+is able to create a record of that fact by creating a new Consent resource.
+This resource will store all the relevant information about the relationship
+between the three parties, and will eventually contain additional information
+for how the User can prove that it consents to each individual transfer in the
+future.
+
+This phase consists exclusively of the DFSP requesting that a new consent be
+created. This request must be conveyed both to the PISP itself and the Auth
+service which will be the record of trust for these resources.
+
+![Grant consent](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/mojaloop/pisp/master/docs/linking/4-grant-consent.puml)
+
+## 1.6. Credential registration
+
+Once the consent resource has been created, the PISP will attempt to establish
+with the network the credential that should be used to verify that a user has
+given consent for each transfer in the future.
+
+This will be done by storing a FIDO credential (e.g., a public key) on the Auth
+service inside the consent resource. When future transfers are proposed, we will
+require that those transfers be digitally signed by the FIDO credential (in this
+case, the private key) in order to be considered valid.
+
+This credential registration is composed of two phases: requesting a challenge
+and finalizing the signature.
+
+### 1.6.1. Requesting a challenge
+
+In this sub-phase, the PISP requests a challenge from the Auth service, which
+will be returned to the PISP via a `PUT /consents/{ID}` API call.
+
+![Credential registration: Challenge](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/mojaloop/pisp/master/docs/linking/5a-credential-registration.puml)
+
+### 1.6.2. Finalizing the credential
+
+Once the challenge is provided to the PISP, the PISP will generate a new
+credential on the device, digitally sign the challenge, and provide the some new
+information about the credential on the Consent resource:
+
+1. The credential itself (the public component)
+2. A signature of the challenge (to be verified against this public component)
+3. The ID of the credential understood by the device itself
+
+This information is provided back to the Auth service, which then verifies that
+the signature is correct. It then updates the status of the credential to
+`"VERIFIED"`, and notifies both the PISP and the DFSP about these new changes
+to the Consent resource.
+
+![Credential registration: Register](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/mojaloop/pisp/master/docs/linking/5b-credential-registration.puml)
```

Comment:
Right - this is due to the link not being live yet (when we merge, these links will begin to actually work... see how they're pointing to mojaloop/pisp/master)

The point of this was to avoid having to keep these up to date and sync'd. If you want to have a peek, feel free to render them using something like VSCode's PlantUML renderer ?

Reply:
Re the "body" of the `generateChallenge` path... It's there, it's just empty. All of the standard routing headers (source/destination) are expected on every request, but there's no body to the request (it's in the sequence diagrams as `{}`).